test/sql/function/list/list_distinct.test

```diff
@@ -0,0 +1,213 @@
+# name: test/sql/function/list/list_distinct.test
```

Comment:
You fixed an issue where `list_contains` did not properly deal with prior selections: #3463

Could you test the same here as well? Just a few tests with a WHERE clause